### PR TITLE
Raise error eagerly on bad unitary synthesis method

### LINKED ISSUE #11355: specifying nonexistent synthesis plugin does not raise error
### Environment

- **Qiskit Terra version**: 0.45.1
- **Python version**: 3.10.12
- **Operating system**: Arch Linux

### What is happening?

`generate_preset_pass_manager` does not raise an error for a nonexistent synthesis plugin, even though it does for a nonexistent stage plugin.

### How can we reproduce the issue?

```python
from qiskit.providers.fake_provider import FakeBelem
from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager

backend = FakeBelem()
pass_manager = generate_preset_pass_manager(
    optimization_level=3, backend=backend, unitary_synthesis_method="a"
)
```

### What should happen?

This does not raise an error. Instead, an error is raised when the pass manager is run. This is inconsistent with the transpiler stage plugins. For example, this raises an error:
```python
from qiskit.providers.fake_provider import FakeBelem
from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager

backend = FakeBelem()
pass_manager = generate_preset_pass_manager(
    optimization_level=3, backend=backend, routing_method="a"
)
```
```
TranspilerError: 'Invalid plugin name a for stage routing'
```

### Any suggestions?

The error should be raised when the pass manager is created, not when it is run.

### PR DESCRIPTION
**Pull request by rupeshknn**
### Summary
`generate_preset_pass_manager` now raises an error the pass manager is created, not when it is run.

Fixes #11355

### CODE REVIEW

**Review by kevinsung — CHANGES_REQUESTED** [review]
Thanks! Please add a test.

**Comment by rupeshknn** [comment]
Do you have a recommendation for where the test should be located? I couldn't find test files (within synthesis) that check whether a particular error was raised.

 Or I could add one where the pass manager is tested. https://github.com/Qiskit/qiskit/blob/main/test/python/transpiler/test_preset_passmanagers.py

**Comment by rupeshknn** [comment]
I've added the test to `test/python/transpiler/test_stage_plugin.py:`

**Review by kevinsung — DISMISSED** [review]
Thank you!

File: releasenotes/notes/move-synthesis-plugin-error-61e3683bf5a0c225.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    fixes issue #11355. If a nonexistent synthesis plugin is passed, `generate_preset_pass_manager` now raises an error 
+    the pass manager is created, not when it is run.
```
Comment (kevinsung) [inline releasenotes/notes/move-synthesis-plugin-error-61e3683bf5a0c225.yaml]:
```suggestion
    when the pass manager is created, not when it is run.
```

**Review by kevinsung — DISMISSED** [review]

**Review by jakelishman — APPROVED** [review]
Thank you!

**Pull request merged**